src/utils/access_log.ts

```diff
@@ -73,32 +96,75 @@ export const storeSpidLogs = (
     E.map(item => ({
       ...spidLogMsg,
       ...item
-    })),
-    E.fold(
-      err =>
-        TE.left(new Error(`StoreSpidLogs|ERROR=Cannot encrypt payload|${err}`)),
-      (encryptedBlobItem: SpidBlobItem) =>
-        pipe(
-          encryptedBlobItem,
-          t.exact(SpidBlobItem).decode,
-          TE.fromEither,
-          TE.mapLeft(
-            errs =>
-              new Error(
-                `StoreSpidLogs|ERROR=Cannot decode payload|ERROR_DETAILS=${readableReport(
-                  errs
-                )}`
-              )
-          ),
-          TE.chain(spidBlobItem =>
-            upsertBlobFromObject(
-              blobService,
-              containerName,
-              `${spidBlobItem.spidRequestId}-${spidLogMsg.createdAtDay}-${spidLogMsg.fiscalCode}.json`,
-              spidBlobItem
-            )
-          )
-        )
-    )
+    }))
+  );
+};
+
+// Create a writer for azure storage
+export const createAzureStorageAccessLogWriter = (
+  blobService: BlobService,
+  containerName: string
+): AccessLogWriter => (
+  encryptedBlobItem: SpidBlobItem,
+  blobName: string
+): ReturnType<AccessLogWriter> =>
+  pipe(
+    upsertBlobFromObject(
+      blobService,
+      containerName,
+      blobName,
+      encryptedBlobItem
+    ),
+    TE.map(_ => void 0)
+  );
+
+// Create a writer for AWS S3
+export const createAwsS3AccessLogWriter = (
+  s3Service: S3Client,
+  bucketName: string
+): AccessLogWriter => (
+  encryptedBlobItem: SpidBlobItem,
+  blobName: string
+): ReturnType<AccessLogWriter> =>
+  pipe(
+    TE.tryCatch(
+      async () =>
+        await s3Service.send(
+          new PutObjectCommand({
+            Body: JSON.stringify(encryptedBlobItem),
+            Bucket: bucketName,
+            Key: blobName
+          })
+        ),
+      E.toError
+    ),
+
+    TE.map(_ => void 0)
   );
+// Create a writer for a given kind
+export const createAccessLogWriter = (
+  storageConfig: SpidLogsStorageConfiguration
+): AccessLogWriter => {
+  if (storageConfig.SPID_LOGS_STORAGE_KIND === "azurestorage") {
+    return createAzureStorageAccessLogWriter(
+      createBlobService(storageConfig.SPID_LOGS_STORAGE_CONNECTION_STRING),
+      storageConfig.SPID_LOGS_STORAGE_CONTAINER_NAME
+    );
+  } else if (storageConfig.SPID_LOGS_STORAGE_KIND === "awss3") {
+    return createAwsS3AccessLogWriter(
+      new S3Client({
+        endpoint: storageConfig.SPID_LOGS_STORAGE_ENDPOINT?.href,
+        region: "us-east-1"
```

Comment:
```suggestion
```

Hardcoding a region would prevent this micro service to work without change in many use cases.

Moreover, for the specific case, PND is not deployed in `us-eaast-1` region.

If I am not mistaken, the library should automatically get the proper region using the `AWS_REGION` environment variable.